src/hdf5/HDF5IO.cpp

```diff
@@ -377,6 +386,49 @@ Status HDF5IO::createStringDataSet(const std::string& path,
   return Status::Success;
 }
 
+Status HDF5IO::startRecording()
+{
+  if (!opened)
+    return Status::Failure;
+
+  if (!disableSWMRMode) {
+    herr_t status = H5Fstart_swmr_write(this->file->getId());
+    return checkStatus(status);
+  }
+  return Status::Success;
+}
+
+Status HDF5IO::stopRecording()
+{
+  // if SWMR mode is disabled, stopping the recording will leave the file open
+  if (!disableSWMRMode) {
+    close();
+  }
+  return Status::Success;
+}
+
+bool HDF5IO::canModifyObjects()
+{
+  if (!opened)
+    return false;
+
+  // Get the file access intent
+  unsigned int intent;
+  herr_t status = H5Fget_intent(this->file->getId(), &intent);
+  if (status < 0) {
+    return false;  // We could not access the file so modifying objects is not
+                   // going to work
+  }
+
+  // Check if SWMR mode is enabled
+  if (intent & (H5F_ACC_SWMR_READ | H5F_ACC_SWMR_WRITE)) {
+    return false;  // File is in SWMR mode
+  } else {
+    return true;  // File is not in SWMR mode
+  }
+  return true;
+}
+
```

Comment:
I think the logic in this function could be simplified to reduce the number of return statements. I think this should work:

```suggestion
{
  // check if we are in swmr mode
  bool smwr_mode = False
  unsigned int intent;
  herr_t status = H5Fget_intent(this->file->getId(), &intent);
  bool status_ok = (status >= 0)
  if(status_ok)
  {
      swmr_mode = (intent & (H5F_ACC_SWMR_READ | H5F_ACC_SWMR_WRITE))
  }
  // if the file is opened and we are not in swmr mode then we can modify objects
  return opended && status_ok && !swmr_mode
}
```

What do you think?

Reply:
I think we still need the early return for `!opened` because `this->file` will be a `nullptr`, but otherwise that works!